msmart/device/AC/command.py

```diff
@@ -116,7 +116,9 @@ def payload(self) -> bytes:
 
         # Build target temp and mode byte
         fractional, integral = math.modf(self.target_temperature)
-        temperature = (int(integral) & 0xF) | (0x10 if fractional > 0 else 0)
+        temperature2 = (int(integral) - 12) & 0x1F

```

Comment:
Based on the Lua code, `temperature2` should be 0 when the target temp is between 17 and 30,

There's also a condition to support an even wider range
```lua
    if (keyP["temperature"] < 17 or keyP["temperature"] > 30) then
      if((keyP["temperature"] >= 13 and keyP["temperature"] < 17) or (keyP["temperature"] <= 38 and keyP["temperature"] > 30))then
        bodyBytes[18] = bit.bor( bit.band(0x1F, (keyP["temperature"] - 12)), bodyBytes[18])
      else
        bodyBytes[18] = bit.bor( bit.band(0x1F, (keyP["temperature"] + 19)), bodyBytes[18])
      end
    end
```

Reply:
Oh, I actually missed this part from the Lua code. I should definitely look into this more closely.

Reply:
I went ahead and made some of this change. Out of laziness (and inconsistent reference code) I've left out the expanded temperature2 range